### PR TITLE
Add sorting feature for available form list

### PR DESCRIPTION
**Pull request by SatyamBansal**
opendatakit/opendatakit#1370
Added sorting menu to display & sort forms 
ezgif-4-60a6f3b2bd

tested on emulator (API27) & on physical device (Android 7.0)

### CODE REVIEW

**Comment by SatyamBansal** [comment]
runtests

**Comment by jbeorse** [comment]
runtests

**Comment by jbeorse** [comment]
Hi @SatyamBansal. This looks good. My one note: can you change the `compareTo` calls in FormChooserListFragment to `compareToIgnoreCase`. I realize they were `compareTo` in the old version, but I don't see why we would want to organize by case like that.

**Comment by linl33** [comment]
@jbeorse maybe the sorting preference should be stored in the PropertiesSingleton? To prevent the preference of 1 app from contaminating the preference of another. Also to be consistent with how other preferences are stored.

**Comment by jbeorse** [comment]
@linl33 hmm that's a good question. This seems like a preference that is only relevant to this individual page. But, I don't see any harm in storing it in PropertiesSingleton, and if some use for this in another app comes up, it might as well already be there. I know you created a similar ticket for ordering Tables. Is this where you imagine it being used?

**Comment by linl33** [comment]
When I said app, I meant different app names like 'default' and 'coldchain'.

**Comment by jbeorse** [comment]
@linl33 you're completely right. Good catch.

@SatyamBansal please store this preference in PropertiesSingleton where the rest of the preferences are.

**Comment by SatyamBansal** [comment]
@linl33 @jbeorse  i m a little confused about this , as general prooperties are mentioned as syncable so should i put it in general properties or in device properties ?

**Comment by linl33** [comment]
It should be in the device properties, so that the value doesn't get cleared after a sync.

**Comment by SatyamBansal** [comment]
@jbeorse @linl33 Apology for the delay , just finished my exam week , I have replaced the sharedpref with the key in properties singleton , kindly review it

**Comment by wbrunette** [comment]
runtests

**Comment by SatyamBansal** [comment]
Unable to traceback the build error :-( , any help?

**Comment by jbeorse** [comment]
The tests will fail because the build server doesn't know about your androidlibrary pull request. Don't worry. 

I'm reviewing both now.

**Comment by jbeorse** [comment]
runtests

**Comment by jbeorse** [comment]
This looks pretty good to me.

Please fix the whitespace changes in strings.xml. It should only show the new strings you added, without all the new tabs. After that is updated I'll merge.

File: survey_app/src/main/java/org/opendatakit/survey/fragments/FormChooserListFragment.java [hunk 1]
```diff
@@ -102,17 +177,79 @@
...
+          if (cmp != 0) {
+            return cmp;
+          }
+          cmp = lhs.formDisplaySubtext.compareToIgnoreCase(rhs.formDisplaySubtext);
+          return cmp;
+        }
+      });
+    } else if (sortingOrder.equals(SORT_BY_TABLEID)) {
+
+      Collections.sort(mItems, new Comparator<FormInfo>() {
```
Comment (linl33) [inline survey_app/src/main/java/org/opendatakit/survey/fragments/FormChooserListFragment.java]: It should sort the method parameter `forms` not the instance variable `mItems`. They behave identically here but future changes could break it.

Comment (jbeorse) [reply survey_app/src/main/java/org/opendatakit/survey/fragments/FormChooserListFragment.java]: Good catch

Comment (SatyamBansal) [reply survey_app/src/main/java/org/opendatakit/survey/fragments/FormChooserListFragment.java]: @linl33 thanks for this :+1:

**Comment by SatyamBansal** [comment]
Done with all the changes

**Comment by jbeorse** [comment]
runtests

**Pull request merged**